### PR TITLE
support ssh-agent on windows

### PR DESCRIPTION
**Pull request by ddding**
support ssh-agent on windows named pipe \\.\pipe\open-ssh-agent

### CODE REVIEW

File: tssh/agent_windows.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+)
+
+// AGENT_PIPE_ID is the default pipe id for openssh ssh-agent on windows.
+const AGENT_PIPE_ID = `\\.\pipe\openssh-ssh-agent`
+
+func getAgentSigners() []*sshSigner {
+	pipeId := AGENT_PIPE_ID
+	// get enviroment override
+	if socket := os.Getenv("SSH_AUTH_SOCK"); socket != "" {
+		pipeId = AGENT_PIPE_ID
```
Comment (lonnywong) [inline tssh/agent_windows.go:17]: Is it override nothing?

File: tssh/agent_unix.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	"golang.org/x/crypto/ssh/agent"
+)
+
+func getAgentSigners() []*sshSigner {
+	socket := os.Getenv("SSH_AUTH_SOCK")
+	if socket == "" {
+		return nil
+	}
+
```
Comment (lonnywong) [inline tssh/agent_unix.go:17]: Should we use `DialTimeout`?

File: tssh/agent_windows.go (hunk 1 above)
Comment (ddding) [reply tssh/agent_windows.go:17]: ssh-agent could use a different pipe id, which would be passed through SSH_AUTH_SOCK

File: tssh/agent_unix.go (hunk 2 above)
Comment (ddding) [reply tssh/agent_unix.go:17]: Yeah, that would be better, I'll make a commit for it.

File: tssh/agent_windows.go (hunk 1 above)
Comment (lonnywong) [reply tssh/agent_windows.go:17]: Shouldn’t it be `pathId = socket`?

Comment (ddding) [reply tssh/agent_windows.go:17]: Right, I made a mistake. Thank you.

**Pull request merged**